### ISSUE TITLE
Jooq module

### ISSUE THREAD

**Issue by murdos** [jhipster/jhipster-lite#9235]
Jooq is a nice alternative to JPA.
It would be great if jhlite supports it.

If integration of jooq-generator (as build plugin) is possible with liquibase/flyway, that would be awesome.

**Comment by fabienpuissant** [comment]
Started to work on it 
Configuration seems ok: https://github.com/fabienpuissant/jhipsterlite-jooq/tree/postgres-liquibase

Need to find relevant example to initialize module

Let me know if you use the plugin that way

**Comment by murdos** [comment]
@fabienpuissant : exemple repo looks fine!

**Comment by fabienpuissant** [comment]
Not convinced about what to do as an axample for the initialization 

Only the configuration without any example is enough ?

Or maybe should I keep a CRUD repository example without the infrastructure primary to avoid RestController ?

**Comment by murdos** [comment]
Jooq module(s) should not contain any sample code, only all the setup.
If you want you can also provide an additional sample module, you can look there are already some.

**Comment by murdos** [comment]
Regarding modules organisation, here's a proposal:
- rename existing postgresql to jpa-postgresl, MySQL to jpa-mysql, ...
- have a feature 'jooq', with modules jooq-postgresql, jooq-mysql, ...
- have another feature 'jooq-codegen-liquibase' with modules for postgresql, MySQL
- and have a feature 'jooq-codegen-flyway'

WDYT?

**Comment by fabienpuissant** [comment]
I was thinking about that since hibernate config was loaded by default
Separate the modules this way seems to be a good idea 

But I don't get last two points, I would include testcontainers codegen by default in jooq-xxxxx module no ?

As I understood, jooq need a codegen anyway. So you want to separate the default codegen and the testcontainer codegen in two modules ?

**Comment by murdos** [comment]
> But I don't get last two points, I would include testcontainers codegen by default in jooq-xxxxx module no ?

If you include testcontainers codegen by default in jooq-xxxxx, it means you already have chosen between liquibase and flyway?
Or I'm missing something?

> As I understood, jooq need a codegen anyway. So you want to separate the default codegen and the testcontainer codegen in two modules ?

Indeed, I would separate things. So you can use `jooq-postgresql` module, and manually use the official jooq codegen maven plugin rather than the testcontainers one (that only support postgres, mysql and mariadb: no support for mssql e.g.).

BTW we will have to find a gradle equivalent of the testcontainers codegen maven plugin (in a first step we can declare a dependency to maven to explicitly disable gradle)

**Comment by pascalgrimaud** [comment]
An idea to provide example is to add a small documentation, in a markdown file. It would be enough

**Comment by fabienpuissant** [comment]
> > But I don't get last two points, I would include testcontainers codegen by default in jooq-xxxxx module no ?
> 
> If you include testcontainers codegen by default in jooq-xxxxx, it means you already have chosen between liquibase and flyway? Or I'm missing something?
> 
> > As I understood, jooq need a codegen anyway. So you want to separate the default codegen and the testcontainer codegen in two modules ?
> 
> Indeed, I would separate things. So you can use `jooq-postgresql` module, and manually use the official jooq codegen maven plugin rather than the testcontainers one (that only support postgres, mysql and mariadb: no support for mssql e.g.).
> 
> BTW we will have to find a gradle equivalent of the testcontainers codegen maven plugin (in a first step we can declare a dependency to maven to explicitly disable gradle)

Yeah my idea was to reduce complexity by including testcontainers codegen directly into jooq-postgresql module for example. 

But lets separate different codegens if you think it's needed and I wasn't aware that the testcontainers one is not supporting all  db

**Comment by pascalgrimaud** [comment]
@murdos @fabienpuissant : can it be closed?

**Comment by fabienpuissant** [comment]
There still a lot to do to fully supports jooq

Gradle compatibility

Liquibase integration 

Flyway integration 

Using jooq under test container plugin (for maven ans gradle)

**Comment by murdos** [comment]
@pascalgrimaud : no, there's still some work to do:
- add gradle plugin for jooq
- use https://github.com/testcontainers/testcontainers-jooq-codegen-maven-plugin in order to generate jooq classes without having to start manually a docker container

**Comment by pascalgrimaud** [comment]
In this case, maybe we should split into smaller tickets, and link to this one, which will be the main?
